assets/activation-scripts/activate

```diff
@@ -156,15 +203,19 @@ if [ -n "$FLOX_CMD" ]; then
   set -- "$FLOX_CMD"
 fi
 
-# Set FLOX_ENV as the path by which all flox scripts can make reference to
-# the environment to which they belong. Use this to define the path to the
-# activation scripts directory.
-# TODO: reconcile with CLI which should be setting this. We must override
-#       the value coming from the CLI for now because it won't be set for
-#       container invocations, and it would have the incorrect value for
-#       nested flox activations.
-_FLOX_ENV="$($_dirname -- "${BASH_SOURCE[0]}")"
-export FLOX_ENV="${FLOX_ENV:-$_FLOX_ENV}"
+# Propagate required variables that are documented as exposed.
+export FLOX_ENV="${_FLOX_ENV}"
```

Comment:
I just want to highlight that with these changes we will no longer honor the FLOX_ENV inherited from the environment in any circumstance, so we'll either get the FLOX_ENV implied by the path where the environment was called or the one provided with the `(-e|--env)` flag. That is fine because the CLI will always provide this flag, but just wanted to say that out loud.